src/transport/raw/GroupAddress.h

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @brief
+ *    File contains definitions on how a connection to a peer can be defined.
+ *
+ */
+
+#pragma once
+
+// #include <stdio.h>
+// #include <inet/IPAddress.h>
+// #include <inet/InetInterface.h>
+// #include <lib/core/CHIPConfig.h>
+#include "PeerAddress.h"
+#include <app/util/basic-types.h>
+
+namespace chip {
+namespace Transport {
+
+/**
+ * Describes how a group on a CHIP network can be addressed.
+ * See 2.5.6.2. IPv6 Multicast Address
+ */
+
+class GroupAddress : public PeerAddress
```

Comment:
Is this going to carry more information than `PeerAddress`? If not I would suggest moving the logic below to some `IPAddress::FromGroupId` method and adding `IPAddress::IsMatterMulticast` (similarly to the existing `IPAddress::IsMulticast`).

Reply:
Otherwise we will need to update all the messaging code which already uses `PeerAddress`.

Reply:
It also seems to me that this is a

```cpp
PeerAddress MakeGroupAddress(chip::FabricId fabric, chip::GroupId group);
```

unless we actually need a separate class for type safety. Could you elaborate  if separate class really makes sense or not? I imagine in some cases we can do extra error checking, however in most instances we probably do not care. We could also have things like `IsGroupAddress` and `GetGroupAddressFabricId` and such that take a PeerAddress as an argument.

Reply:
I have replaced the GroupAddress class with a couple of constructor methods in PeerAddress. Though the address will never be TCP, or UDP, in order to accommodate multicast communication with the existing infrastructure, the multicast address should be wrapped inside the PeerAddress.